Bucketeer2Tests/EventInteractorTests.swift

```diff
@@ -0,0 +1,523 @@
+            eventsMaxBatchQueueCount: 3,
+            apiClient: api,
+            eventDao: dao,
+            clock: clock,
+            idGenerator: idGenerator,
+            logger: logger
+        )
+        let listener = MockEventUpdateListener()
+        interactor.set(eventUpdateListener: listener)
+        interactor.sendEvents(completion: { result in
+            switch result {
+            case .success(let success):
+                XCTAssertFalse(success)
+            case .failure:
+                XCTFail()
+            }
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 1)    }
+
+    func testSendEventsForce() throws {
+        let expectation = XCTestExpectation()
+        expectation.assertForOverFulfill = true
+        expectation.expectedFulfillmentCount = 3
+
+        let clock = MockClock(timestamp: 1)
+        let idGenerator = MockIdGenerator(identifier: "id")
+        let addedEvents: [Event] = [.mockEvaluation1]
+        let dao = MockEventDao()
+        try dao.add(events: addedEvents)
+        let api = MockApiClient(registerEventsHandler: { events, completion in
+            XCTAssertEqual(events.count, 1)
+            XCTAssertEqual(events, addedEvents)
+            completion?(.success(.init(data: .init(errors: [:]))))
+            expectation.fulfill()
+        })
+
+        let logger = MockLogger()
+        let interactor = EventInteractor(
+            eventsMaxBatchQueueCount: 3,
+            apiClient: api,
+            eventDao: dao,
+            clock: clock,
+            idGenerator: idGenerator,
+            logger: logger
+        )
+        let listener = MockEventUpdateListener { events in
+            XCTAssertEqual(events.count, 0)
+            expectation.fulfill()
+        }
+        interactor.set(eventUpdateListener: listener)
+        interactor.sendEvents(force: true, completion: { result in
+            switch result {
+            case .success(let success):
+                XCTAssertTrue(success)
+            case .failure:
+                XCTFail()
+            }
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testSendEventsRetriableError() throws {
+        let expectation = XCTestExpectation()
+        expectation.assertForOverFulfill = true
+        expectation.expectedFulfillmentCount = 3
+
+        let clock = MockClock(timestamp: 1)
+        let idGenerator = MockIdGenerator(identifier: "id")
+        let addedEvents: [Event] = [.mockEvaluation1, .mockGoal1]
+        let dao = MockEventDao()
+        try dao.add(events: addedEvents)
+        let api = MockApiClient(registerEventsHandler: { events, completion in
+            XCTAssertEqual(events.count, 2)
+            XCTAssertEqual(events, addedEvents)
+            completion?(.success(.init(data: .init(
+                errors: [
+                    Event.mockEvaluation1.id: .init(retriable: true, message: "message")
```

Comment:
Could you also verify that Event.mockEvaluation1 is still in the DB?

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/test/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractorTest.kt#L617-L648

Reply:
Does this follow your point?

- Verify 2 events before `sendEvents` https://github.com/ca-dp/bucketeer-ios-sdk/pull/85/files#diff-b1760fb1974c08fa44a0e379b736e22d188e107eaa07512c4e4470db068e3de8R488-R489
- Verify that 1 event remains after `sendEvents` https://github.com/ca-dp/bucketeer-ios-sdk/pull/85/files#diff-b1760fb1974c08fa44a0e379b736e22d188e107eaa07512c4e4470db068e3de8R511-R512

Reply:
Yes, it's correct!
Thank you!